selfdrive/car/hyundai/carstate.py

```diff
@@ -45,63 +45,72 @@ def get_can_parser(CP):
     ("CF_Clu_AmpInfo", "CLU11", 0),
     ("CF_Clu_AliveCnt1", "CLU11", 0),
 
-    ("CF_Clu_InhibitD", "CLU15", 0),
-    ("CF_Clu_InhibitP", "CLU15", 0),
-    ("CF_Clu_InhibitN", "CLU15", 0),
-    ("CF_Clu_InhibitR", "CLU15", 0),
-
-    ("CF_Lvr_Gear", "LVR12",0),
-    ("CUR_GR", "TCU12",0),
-
     ("ACCEnable", "TCS13", 0),
     ("ACC_REQ", "TCS13", 0),
+    ("BrakeLight", "TCS13", 0),
     ("DriverBraking", "TCS13", 0),
     ("DriverOverride", "TCS13", 0),
 
     ("ESC_Off_Step", "TCS15", 0),
 
     ("CF_Lvr_GearInf", "LVR11", 0),        #Transmission Gear (0 = N or P, 1-8 = Fwd, 14 = Rev)
 
-    ("CR_Mdps_DrvTq", "MDPS11", 0),
-
     ("CR_Mdps_StrColTq", "MDPS12", 0),
     ("CF_Mdps_ToiActive", "MDPS12", 0),
     ("CF_Mdps_ToiUnavail", "MDPS12", 0),
     ("CF_Mdps_FailStat", "MDPS12", 0),
     ("CR_Mdps_OutTq", "MDPS12", 0),
 
+    ("SAS_Angle", "SAS11", 0),
+    ("SAS_Speed", "SAS11", 0),
+
+    ("MainMode_ACC", "SCC11", 0),
     ("VSetDis", "SCC11", 0),
     ("SCCInfoDisplay", "SCC11", 0),
+    ("ACC_ObjDist", "SCC11", 0),
     ("ACCMode", "SCC12", 1),
-
-    ("SAS_Angle", "SAS11", 0),
-    ("SAS_Speed", "SAS11", 0),
   ]
 
   checks = [
     # address, frequency
     ("MDPS12", 50),
-    ("MDPS11", 100),
     ("TCS15", 10),
-    ("TCS13", 50),
     ("CLU11", 50),
     ("ESP12", 100),
-    ("EMS12", 100),
```

Comment:
Same for this one

Reply:
Same as prev

Reply:
See line 219

Reply:
these cars do not have this message: KIA OPTIMA HYBRID and HYUNDAI IONIQ ELECTRIC /HYBRID. which make sense, they do not have engine ECU signal.
also, they are use the new gear signal
we can do something like this:
`  if CP.carFingerprint in FEATURES["use_elect_gears"]:
    checks.remove(("EMS12", 100))`

Reply:
This isn't the problem, the problem is that the message is being used in an ungated way. What I'm hearing here is that those cars don't have a complete carstate.

Reply:
glad you mention that since all new signals have not added to checks
all signals and checks are gated or removed